renderdoc/replay/entry_points.cpp

```diff
@@ -590,4 +592,82 @@ extern "C" RENDERDOC_API void RENDERDOC_CC RENDERDOC_EndSelfHostCapture(const ch
     return;
 
   rdoc->EndFrameCapture(NULL, NULL);
-}
\ No newline at end of file
+}
+
+namespace Android
+{
+bool IsHostADB(const char *hostname)
+{
+  return !strncmp(hostname, "adb:", 4);
+}
+string adbExecCommand(const string &args)
+{
+  string adbExePath = RenderDoc::Inst().GetConfigSetting("adbExePath");
+  Process::ProcessResult result;
+  Process::LaunchProcess(adbExePath.c_str(), "", args.c_str(), &result);
+  RDCLOG("COMMAND: adb %s", args.c_str());
+  if(result.strStdout.length())
+    // This could be an error (i.e. no package), or just regular output from adb devices.
+    RDCLOG("STDOUT:\n%s", result.strStdout.c_str());
+  return result.strStdout;
+}
+void adbForwardPorts()
+{
+  adbExecCommand(StringFormat::Fmt("forward tcp:%i tcp:%i",
+                                   RenderDoc_RemoteServerPort + RenderDoc_AndroidPortOffset,
+                                   RenderDoc_RemoteServerPort));
+  adbExecCommand(StringFormat::Fmt("forward tcp:%i tcp:%i",
+                                   RenderDoc_FirstTargetControlPort + RenderDoc_AndroidPortOffset,
+                                   RenderDoc_FirstTargetControlPort));
+}
+uint32_t startAndroidPackageForCapture(const char *package)
+{
+  adbExecCommand(string("shell am force-stop ") + package);
+  adbForwardPorts();
+  adbExecCommand("shell setprop debug.vulkan.layers VK_LAYER_RENDERDOC_Capture");
+  adbExecCommand(string("shell monkey -p ") + package + " -c android.intent.category.LAUNCHER 1");
+  Threading::Sleep(
+      5000);    // Let the app pickup the setprop before we turn it back off for replaying.
```

Comment:
One thing is that the UI calls into this function when you hit capture on the UI thread, so this will cause a 5 second "not responding" period for the UI.

I don't think it's something you have to fix in this changelist - the 'wait for debugger' option has exactly the same problem so I'm planning to farm this off to a worker thread. As long as that won't break anything here then it should go away in time, but it's something to be aware of at the moment.